### PR TITLE
Add NAC AP470 Air Purifier

### PR DESCRIPTION
**Pull request by wizu**
Resolves https://github.com/make-all/tuya-local/issues/1356

### CODE REVIEW

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+name: Air purifier
+products:
+  - id: y7iiubdc5sufjvo3
+    name: "NAC AP470 Air Purifier"
+primary_entity:
+  entity: fan
+  translation_key: fan_with_presets
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:7]: When using this key, your preset_mode values should be mapped to keys from the translation file https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/translations/en.json#L170

The values currently in there were taken from presets that used to be listed in HA source code, but are no longer there, and I guess in future will become default translations for fan when they get around to translating the fan platform fully. "sleep" is the only direct mapping, but "smart" should be a good substitute for Auto, and "baby", "normal", "fresh", "strong"  could be a progression for the other levels.

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    dps:
+      - id: 2
+        type: integer
+        name: sensor
+        class: measurement
+        unit: ugm3
+  - entity: sensor
+    name: Filter left days
+    icon: "mdi:air-filter"
+    category: diagnostic
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:44]: `class: duration` might apply here (I'm not sure it changes anything beyond hours though)

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    dps:
+      - id: 16
+        type: integer
+        name: sensor
+        unit: "d"
+  - entity: sensor
+    name: Total time
+    icon: "mdi:air-filter"
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:52]: again, `class: duration`

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        unit: "h"
+  - entity: switch
+    name: Ionizer
+    icon: "mdi:atom-variant"
+    category: config
+    dps:
+      - id: 6
+        type: boolean
+        name: switch
+        mapping:
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:67]: There is no mapping here, it should just be removed

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        name: lock
+  - entity: switch
+    name: Light
+    category: config
+    icon: "mdi:lightbulb"
+    dps:
+      - id: 8
+        type: boolean
+        name: switch
+        mapping:
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:86]: Again, no mapping needed

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+          - dps_val: false
+  - entity: switch
+    name: UV lamp
+    category: config
+    icon: "mdi:shield-bug"
+    dps:
+      - id: 9
+        type: boolean
+        name: switch
+        mapping:
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:97]: no mapping needed

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+          - dps_val: true
+          - dps_val: false
+  - entity: switch
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:100]: Is this a switch, or a binary_sensor (an indicator to change/clean the filter)

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    name: Filter clean
+    category: config
+    icon: "mdi:air-filter"
+    dps:
+      - id: 101
+        type: boolean
+        name: switch
+        mapping:
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:108]: no mapping needed

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  - entity: select
+    name: Timer
+    icon: "mdi:timer"
+    category: config
+    dps:
+      - id: 19
+        name: option
+        type: string
+        mapping:
+          - dps_val: cancle
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:128]: Just confirming, the typo is in the actual device, not your config?

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            value: "8 hours"
+  - entity: sensor
+    name: Timer time left
+    icon: "mdi:timer"
+    category: diagnostic
+    dps:
+      - id: 20
+        type: integer
+        name: sensor
+        unit: "m"
```
Comment (make-all) [inline custom_components/tuya_local/devices/nac_ap470_purifier.yaml:146]: min for minutes (m is metres)

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml (hunk 1 above)
Comment (wizu) [reply custom_components/tuya_local/devices/nac_ap470_purifier.yaml]: I've removed translation_key: fan_with_presets, and will leave names as it was. They are the same as in tuya app.

File: custom_components/tuya_local/devices/nac_ap470_purifier.yaml (hunk 7 above)
Comment (wizu) [reply custom_components/tuya_local/devices/nac_ap470_purifier.yaml:92]: In tuya app this is switch. On some interval this switch automaticaly turns on and there is an icon on display. Tu turn it off you have to switch off this in app.

**Pull request merged**